app_dart/lib/src/request_handlers/refresh_cirrus_status.dart

```diff
@@ -37,60 +42,91 @@ class RefreshCirrusStatus extends ApiRequestHandler<Body> {
     Config config,
     AuthenticationProvider authenticationProvider, {
     @visibleForTesting DatastoreServiceProvider datastoreProvider,
+    @visibleForTesting
+        this.branchHttpClientProvider = Providers.freshHttpClient,
+    @visibleForTesting this.gitHubBackoffCalculator = twoSecondLinearBackoff,
   })  : datastoreProvider =
             datastoreProvider ?? DatastoreService.defaultProvider,
+        assert(branchHttpClientProvider != null),
+        assert(gitHubBackoffCalculator != null),
         super(config: config, authenticationProvider: authenticationProvider);
 
   final DatastoreServiceProvider datastoreProvider;
+  final HttpClientProvider branchHttpClientProvider;
+  final GitHubBackoffCalculator gitHubBackoffCalculator;
 
   @override
   Future<Body> get() async {
     final DatastoreService datastore = datastoreProvider();
     final GraphQLClient client = await config.createCirrusGraphQLClient();
+    final GitHub github = await config.createGitHubClient();
+    const RepositorySlug slug = RepositorySlug('flutter', 'flutter');
+    final Stream<Branch> branches = github.repositories.listBranches(slug);
+    final List<String> regExps = await loadBranchRegExps(
```

Comment:
I wonder if this logic can be abstracted to a utility function that can be used everywhere(I've seen this logic replicated in several places).

Ideally something like:

List<Branch> branches = await getSupportedBranches();
for (Branch branch in branches) { ...

Reply:
Rewrite this part as suggested. Now multiple APIs call this shared logic defined in utils.dart. Additionally merged the /request_handler/utils.dart with /foundation/utils.dart.